File: app/src/main/java/com/qwy/designpattern/lagou/singletonpattern/java/SingletonPattern05.java
package com.qwy.designpattern.lagou.singletonpattern.java;

/**
 * 静态内部类
 *
 */

public class SingletonPattern05 {


    private static class SingleHolder {
        private static final SingletonPattern05 INSTANCE = new SingletonPattern05();
    }


    public SingletonPattern05() {
    }


    public static final SingletonPattern05 getInstance() {
        return SingleHolder.INSTANCE;
    }
}

File: app/src/main/java/com/qwy/designpattern/lagou/singletonpattern/kotlin/SingletonPattern05.kt
package com.qwy.designpattern.lagou.singletonpattern.kotlin

/**
 * 静态内部类式
 */
class SingletonPattern05 {

    companion object {
        val instance = SingletonHolder.holder
    }


    private object SingletonHolder {
        val holder = SingletonPattern05()
    }


}